inst/private/python_ipc_system.m

```diff
@@ -106,14 +107,19 @@
   else
     %% Generate a temp .py file then execute it with system()
     % can be useful for debugging, or if "python -c" fails for you
-    fname = 'tmp_python_cmd.py';
-    fd = fopen(fname, 'w');
+    if (isempty (tmpfilename))
+      tmpfilename = [tempname() '_octsympytmp.py'];
+      if (verbose)
+        disp (['This session will use the temp file "' tmpfilename '"'])
+      end
+    end
+    fd = fopen (tmpfilename, 'w');
     fprintf(fd, '# temporary autogenerated code\n\n');
     % we just added two more lines at the top
     info.prelines = info.prelines + 2;
     fputs(fd, bigs);
     fclose(fd);
```

Comment:
I think there's a potential race condition here due to the use of `tempname`. In the best case, a race condition would merely cause inconvenience by creating nondeterministic test failure. In the worst case, it could be a security hole. 

Quoting the [python documentation](https://docs.python.org/3/library/tempfile.html),

```
A historical way to create temporary files was to first generate a file name with the mktemp() function and then create a file using this name. Unfortunately this is not secure, because a different process may create a file with this name in the time between the call to mktemp() and the subsequent attempt to create the file by the first process. The solution is to combine the two steps and create the file immediately. This approach is used by mkstemp() and the other functions described above.
```

Unfortunately, while octave do provide `tmpfile` and `mkstemp` to create temp file w/o race condition, matlab doesn't, unlike most modern programming languages.

I would suggest creating temp file using `mkstemp` in octave and use `tempfile.mkstemp` in python otherwise.

Reply:
yeah, I noticed that about Matlab, which is why I didn't do `mkstemp`.   We do not reliably have a Python connect at this point.

I think best we could do is Octave-dependent code, using the proper secure approach and then tempname on Matlab...   I haven't tested on Matlab in a long time so its not worth much effort.

File an issue for this for a future improvement?

Reply:
The other thing to keep in mind is that this `systmpfile` stuff is basically for debugging, not production use (although we do call it in our CI)

We could probably just deprecate this code, especially if/when Pythonic matures.

Reply:
Sounds good to me. So for now, we workaround this by using Octave-dependent code and deprecate this code in the future.

I believe we should also deprecate code using `system` in the future as well. It is well-known that passing non-constant command string (those depends on user inputs) to shell is dangerous because escaping string is error-prone. There has been an unlimited supply of arbitrary code execution bugs caused by incorrect escaping.

I think the octave documentation should be amended as well to state these well-known risks for `tempname` and `system`.

Reply:
Good point!  Let's file some new issues from this discussion:
 * `help sympref` should mention the inherent insecurity of these alternative IPC mechanisms.
 * systmpfile could use some `if is_octave()`  (see other code, something like a variable OCTAVE_VER is defined or not).

I suspect a patch to upstream Octave for tempname and system would be accepted.